### DISCUSSION TITLE
Make current and previous versions download available

### DISCUSSION THREAD

**Discussion by HaGGi13** [rfennell/AzurePipelines#1861] [Ideas]
Hello,

Current we are faced with issues to install the latest available package version (_v4.16.13 -  published 18.3.2024, 21:51:00_) out of the marketplace.
So I wanted to download and install the package manually. Unfortunately, the lastest available version is from 2019.

> _Side note - here is the error we got:_
> ```
> Task definition visible rule 'getPRDetailsRepositoryId = ''' is invalid. It should be of format "A <validconditionhere> B" for Task with ID '7b0a5887-75c4-4ade-a915-f92a93c934cb'.
> ```

This brought me to the fact to suggest, to make all on marketplace published versions available to download via the GitHub page like other extensions do.

If this is already the case and I just didn't found it, I am super sorry and please to get pointed to.
In this case, it should be somewhere referenced to, to make it more obvious.

Thank you in advance and I am happy for any reaction on this.

Chris

**Comment by rfennell** [comment]
From the GUID I assume the task you are using is the XplatRelease notes, I ask as there are a number of them in this repository, and that is the reason why up to now I have not added VSIX packages as releases in the repo. I felt the different packages, with different release numbers in a single repo would be confusing if published as Git releases

So what I have done is make the Azure DevOps instance where I build and test the extensions public, and you can download the VSIX packages from retained pipelines e.g. [Xplat release notes public builds](https://dev.azure.com/richardfennell/GitHub/_build?definitionId=85&branchFilter=473%2C473%2C473%2C473%2C473%2C473%2C473%2C473%2C473%2C473%2C473)

With hindsight I think put each extension in it's own repo would have been a better choice, maybe a change I should make, as it would help address your issue, and also make Dependabot usage easier (at present it generates some nasty PRs to resolve)

**Comment by HaGGi13** [comment]
Hi @rfennell, thanks a lot for your answer.

I see, I didn't realize before that you were using a mono repo approach. Yes, that makes it difficult to have different versions for each of the components it contains.
Indeed it would be easier to version each component if it would be located in its own repository. By doing so, you could use Git Actions to build and publish your packages with specific versions and adding the version as Git tag to the history.

Regards my initial question.
Thanks for sharing the pipeline link. I downloaded the package and tried to install it, but unfortunately I got the same error.
![image](https://github.com/rfennell/AzurePipelines/assets/4867233/9932c8f4-c930-49ff-9b36-972a448de964)

I'll search in the issues if something similar was already reported somewhen.


Thank you.
